proposals/lip-0053.md

```diff
@@ -252,178 +479,192 @@ crossChainUpdateTransactionParams = {
+    if len(trs.params.certificate) > 0:
+        verifyCertificate(trs)
+    if len(trs.params.activeValidatorsUpdate) > 0 or trs.params.newCertificateThreshold > 0:
+        verifyValidatorsUpdate(trs)
+    if trs.params.inboxUpdate is not empty: # An empty object has all properties set to their default values.
+        verifyChannel(trs)
+```
 
-#### Common Processing
+##### Execution
 
-For CCU transactions posted on the mainchain or on sidechains, once the specific execution steps described above have been completed, the following is done in the given order:
+```python
+def execute(trs: Transaction) -> None:
+    # Verify certificate signature. We do it here because if it fails, the transaction fails rather than being invalid.
+    verifyCertificateSignature(trs)
+    # Process cross-chain messages in inbox update.
+    crossChainMessages = trs.params.inboxUpdate.crossChainMessages
+
+    # First process basic checks for all CCMs.
+    for ccmBytes in crossChainMessages:
+        # Verify general format. Past this point, we can access ccm root properties.
+        try:
+            validateFormat(ccm)
+            # The CCM must come be directed to the sidechain, unless it was bounced on the mainchain.
+            if ownChainAccount.ID != ccm.receivingChainID and ccm.status != CCM_STATUS_CHANNEL_UNAVAILABLE:
+                raise Exception("CCM is not directed to the sidechain.")
+            elif ownChainAccount.ID != ccm.sendingChainID and ccm.status == CCM_STATUS_CHANNEL_UNAVAILABLE:
+                raise Exception("Bounced CCM was not coming from the sidechain.")
+        except:
+            terminateChain(trs.params.sendingChainID)
+            # In this case, we do not even update the chain account with the new certificate.
+            return
+    
+    for ccmBytes in crossChainMessages:
+        try:
+            ccm = decode(crossChainMessageSchema, ccmBytes)
+            # Modules can verify the CCM. In particular, this calls the interoperability verifyCrossChainMessage defined above.
+            for each module mdl for which verifyCrossChainMessage exists:
+                mdl.verifyCrossChainMessage(trs, ccm)
+        except:
+            # Chain was already terminated.
+            if terminatedStateAccount(chainID) exists:
+                continue
+                
+            # Terminate sending chain.
+            terminateChain(ccm.sendingChainID)
+            ccmID = SHA-256(ccmBytes)
+            
+            emitEvent(
+                moduleID = MODULE_ID_INTEROPERABILITY,
+                typeID = TYPE_ID_CCM_PROCESSED,
+                data = {"ccmID": ccmID, "status": CCM_PROCESSED_STATUS_COMMAND_DISCARDED},
+                topics = [ccm.sendingChainID, ccm.receivingChainID]
+            )
+            # Notice that, even if the sending chain has been terminated, 
+            # future CCMs from other chains can still be processed.
+            continue
```

Comment:
Don't I still need to call `appendToInboxTree(trs.params.sendingChainID, ccmBytes)` in this case?

Reply:
indeed, fixed

Reply:
Looks good now, I resolve the comment.